### PR TITLE
For #1147: Each user should see the status of his/her resume at /join.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1147:
- Created test in `TkJoinTest` which tests if user is seeing page telling that he /she already has a mentor;
- Created test in `TkJoinTest` which tests if user is seeing resume on page, it he / she already has a resume.
- Added in puzzle the info about ticket implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1165](https://github.com/zerocracy/farm/issues/1165) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1165](https://github.com/zerocracy/farm/issues/1165) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1690375920); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 1]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+    /**
+     * {@link TkJoin} can show that user already has a mentor. TkJoin must
+     * show a message to user when he or she tries to access
+     * <code>/join</code> but already has a mentor, which means that the user
+     * has already joined or asked for menthor for joining.
+     * @throws IOException If something goes wrong accessing page.
+     */
+    @Test
+    @Ignore
+    public void showThatUserAlreadyHasMentor() throws IOException {
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso `show` -> `shows` (https://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names)

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 2]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+
+    /**
+     * {@link TkJoin} can show resume if user already applied. TkJoin must
+     * show user's resume, if there is one, when user tries to access
+     * <code>/join</code> endpoint.
+     * @throws IOException If something goes wrong accessing page.
+     */
+    @Test
+    @Ignore
+    public void showResumeIfAlreadyApplied() throws Exception {
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso `show` -> `shows` (https://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names)

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 3]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
             )
         );
     }
+
+    /**
+     * {@link TkJoin} can show that user already has a mentor. TkJoin must
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso there should be a new line after the first sentence of javadoc (this is usually the summary).

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 4]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+                new IterableOf<String>(
+                    "User",
+                    "is already your mentor, no need to join again"
+                )
+            )
+        );
+    }
+
+    /**
+     * {@link TkJoin} can show resume if user already applied. TkJoin must
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso there should be a new line after the first sentence of javadoc (this is usually the summary).

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 5]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+     * show user's resume, if there is one, when user tries to access
+     * <code>/join</code> endpoint.
+     * @throws IOException If something goes wrong accessing page.
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso there shouldn't be a comma at the end of tag descriptions like throws, return or param (unless you have more than 1 sentence)

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 6]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
             )
         );
     }
+
+    /**
+     * {@link TkJoin} can show that user already has a mentor. TkJoin must
+     * show a message to user when he or she tries to access
+     * <code>/join</code> but already has a mentor, which means that the user
+     * has already joined or asked for menthor for joining.
+     * @throws IOException If something goes wrong accessing page.
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso there shouldn't be a comma at the end of tag descriptions like throws, return or param (unless you have more than 1 sentence)

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 7]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+     */
+    @Test
+    @Ignore
+    public void showThatUserAlreadyHasMentor() throws IOException {
+        final Farm farm = new PropsFarm(new FkFarm());
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso those two test methods are exactly the same except the assertion was that the intent?

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 8]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+        final People people = new People(farm).bootstrap();
+        final String uid = "yegor256";
+        people.touch(uid);
+        people.apply(uid, new Date());
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso I think here you would need also `invite` after the apply

File: src/test/java/com/zerocracy/tk/TkJoinTest.java [hunk 9]
```diff
@@ -127,4 +131,63 @@ public void acceptIfNeverApplied() throws Exception {
...
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new TkApp(farm).act(
+                    new RqWithUser(farm, new RqFake("GET", "/join"))
+                )
+            ).printBody(),
+            new StringContainsInOrder(
+                new IterableOf<String>(
+                    "User",
```
Comment (krzyk) [inline src/test/java/com/zerocracy/tk/TkJoinTest.java]: @paulodamaso and here you should check the username that invited given person

File: src/test/java/com/zerocracy/tk/TkJoinTest.java (hunk 7 above)
Comment (paulodamaso) [reply src/test/java/com/zerocracy/tk/TkJoinTest.java]: @krzyk Yes, we are testing two behaviors: one if user with mentor is trying to access '/join' endpoint, and another if user with resume is accessing '/join'. Looks like that is possible to a user do not have a resume, so it the page should show resume only if it exists.

**Comment by paulodamaso** [comment]
@krzyk Thanks for the review, made some corrections, please take a look,

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1165](https://github.com/zerocracy/farm/issues/1165) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Pull request merged**